pkg/framework/spi/crypto/api.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package crypto
+
+import "github.com/google/tink/go/keyset"
+
+// Provider interface for Crypto ctx
+type Provider interface {
+}
+
+// Creator method to create new pluggable Crypto service
+type Creator func(prov Provider) (Crypto, error)
+
+// Crypto is an interface defining a pluggable Crypto operations provider to Aries agents.
+// It's only responsible for pure crypto operations and nothing more. Envelope building and keys management are managed
+// in other interfaces
+type Crypto interface {
+	// Encrypt will encrypt msg and aad using a matching AEAD primitive in kh
+	// returns:
+	// 		cipherText in []byte
+	//		nonce in []byte
+	//		error in case of errors during encryption
+	Encrypt(msg, aad []byte, kh *keyset.Handle) ([]byte, []byte, error)
```

Comment:
We lose flexibility by exposing tink to implementations via the API. Plugin architectures are generally meant to be flexible.

Reply:
Note: here is the definition of keyset.Handle: https://github.com/google/tink/blob/090075d031a55da56bf0551b2e1495d19e38c000/go/keyset/handle.go#L33

Reply:
This implementation almost suggests that tink, itself, is being used as the SPI.

i.e., this feels like a 50/50 solution right now. You didn't directly use tink as the Crypto SPI but used some definitions (and particularly a Key Handle - the essence of a Crypto SPI, I would think).

Reply:
"50/50 solution" is spot on.

Reply:
yes.. the intention is not to use Tink as an SPI, but to use its services in our pluggable Crypto and KMS APIs